keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1184 @@
+Once the new version is confirmed to work, all applications can be updated to
+point to it.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None determined so far.
+
+##### Unit tests
+
+All changes will have unit test coverage.
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+(I'm not yet sure precisely which packages will need to be adjusted.)
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+The following scenarios will need to be tested using integration tests:
+
+* kube-apiserver forbids creation and update of a ClusterTrustBundle targeted if
+  the requester doesn't have the `entrust` verb on the corresponding signer.
+
+* kube-apiserver correctly connects to validating, mutating, and conversion
+  webhook backends that use a ClusterTrustBundle for managing their trust
+  anchors, and can follow a CA rotation in the corresponding ClusterTrustBundle.
+
+* kube-apiserver correctl connects to aggregated API server backends that use a
+  ClusterTrustBundle for managing their trust anchors.
+
+Additionally, it would be nice to test the following kubelet behavior in integration tests if feasible:
+
+* Kubelet correctly writes (and updates) the content of a ClusterTrustBundle
+  named in the pod spec into the container fileystem.
+
+* Kubelet blocks pod startup if the pod spec refers to a ClusterTrustBundle that
+  doesn't exist.
+
+* Kubelet blocks pod startup if the pod spec refers to a ClusterTrustBundle with
+  an empty `.spec.pemTrustAnchors` field.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+The following end-to-end tests will be needed:
```

Comment:
if the kube-apiserver is expected to auto-reload trust based on the clustertrustbundles, that functionality should be tested as well.

Reply:
We've settled on this not being part of the alpha functionality, but will be revisited for beta.